grammar.js

```diff
@@ -698,6 +699,8 @@ module.exports = grammar({
       field('return_type', $._type)
     )),
 
+    _structural_type: $ => $.template_body,
```

Comment:
Not sure if it shouldn't be just inlined there.

Reply:
For later purpose of highlighting or folding, I wonder if this should be a visible node, that is `structural_type` without the underscore.

Reply:
For folding - sure, and this can just be aliases `_structural_type: $=>alias($.template_body, $.structural_type)` if this is the correct syntax

Reply:
Right, there is also another way. We could make it an alias like `alias($.template_body, 'structural_type')`. 

So instead of having it a visible node which would render as:
```
(compilation_unit
  (type_definition (type_identifier)
    (structural_type
      (template_body
        (function_declaration 
          (identifier)
          (parameters)
          (type_identifier)
        )
      )
    )
  )
)
```
we would have:
```
(compilation_unit
  (type_definition (type_identifier)
      (structural_type
        (function_declaration 
          (identifier)
          (parameters)
          (type_identifier)
        )
    )
  )
)
```
wdyt?

Reply:
I wonder if actually having the template_body there is better, because we can do generic highlights (like, "all function declarations in template_body should be marked with `method`, not `function`)

That said, I don't see any such queries there yet, and we can change them later.

So for now - happy to go with 

```
      (structural_type
        (function_declaration 
          (identifier)
          (parameters)
          (type_identifier)
        )
    )
```

because it reduces nesting in the tree.